### ISSUE TITLE
Segmentation fault: cuDNN6, cuda8 and Theano 0.9.0

### ISSUE THREAD

**Issue by loretoparisi** [Theano/Theano#6202]
I'm running `Theano` with `nvidia-docker` where `FROM nvidia/cuda:8.0-cudnn6-devel-ubuntu16.04`:

```
root@d842fc00a358:~/GRU4Rec/examples/rsc15# python -c "import theano; print( theano.__version__)"
0.9.0.dev-c697eeab84e5b8a74908da654b66ec9eca4f1291
```

My `.theanorc` is 

```
[global]
floatX = float64
device = cuda0

[lib]
cnmem = 1

[nvcc]
fastmath = True
```

and my gpu is

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 367.57                 Driver Version: 367.57                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GRID K520           Off  | 0000:00:03.0     Off |                  N/A |
| N/A   38C    P8    17W / 125W |      0MiB /  4036MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

If I run the gpu test script [here](http://deeplearning.net/software/theano/tutorial/using_gpu.html#testing-theano-with-gpu) I get this warning:

```
# python gpu_test.py 
/root/yes/lib/python3.5/site-packages/theano/gpuarray/dnn.py:135: UserWarning: Your cuDNN version is more recent than Theano. If you encounter problems, try updating Theano or downgrading cuDNN to version 5.1.
  warnings.warn("Your cuDNN version is more recent than "
Using cuDNN version 6021 on context None
Mapped name None to device cuda0: GRID K520 (0000:00:03.0)
[GpuElemwise{exp,no_inplace}(<GpuArrayType<None>(float64, (False,))>), HostFromGpu(gpuarray)(GpuElemwise{exp,no_inplace}.0)]
Looping 1000 times took 1.100811 seconds
Result is [ 1.23178032  1.61879341  1.52278065 ...,  2.20771815  2.29967753
  1.62323285]
Used the gpu
```

**NOTE**

- The `cpu` version works i.e. no segmentation fault while training.
- My theano configuration properties with gpu are [here](https://gist.github.com/loretoparisi/785fdbf0b88a36fd1ceeb68ffd48cd97)
- The full Dockerfile is [here](https://github.com/loretoparisi/docker/blob/master/theanogru/Dockerfile).
- `cuda` Dockerfile options are [here](https://hub.docker.com/r/nvidia/cuda/).

**Comment by loretoparisi** [comment]
**[UPDATE]**
After downgrading to `cudnn5` so using `FROM nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04
`
I do not have the warning anymore:

```
root@2b71143cedda:~# python gpu_test.py 
Using cuDNN version 5110 on context None
Mapped name None to device cuda0: GRID K520 (0000:00:03.0)
[GpuElemwise{exp,no_inplace}(<GpuArrayType<None>(float64, (False,))>), HostFromGpu(gpuarray)(GpuElemwise{exp,no_inplace}.0)]
Looping 1000 times took 0.909592 seconds
Result is [ 1.23178032  1.61879341  1.52278065 ...,  2.20771815  2.29967753
  1.62323285]
Used the gpu
```

Not sure if this is causing the segmentation fault btw.

**Comment by loretoparisi** [comment]
**[UPDATE]**
I'm not sure of this, but after moving to `python3` and `pip3` rather than `anaconda`, and building `libgparray` from scratch I was able to train withtout the segfault.
The updated Dockerfile is [here](https://github.com/loretoparisi/docker/blob/master/theanogru/Dockerfile).

If confirmed, this issue can be closed.

**Comment by PumpkingWok** [comment]
I have the same segmentation fault problem, I'm using Theano 0.9, cuda 8 and cuDNN 6, I also tried with cuDNN 5.1 but nothing.

**Comment by loretoparisi** [comment]
@PumpkingWok If you try the Dockerfile above or the same packages install on `Ubuntu16.04LTS`, `cuda8.0` and `cuDNN5.0` it should work as for my last attempt.

**Comment by botev** [comment]
@loretoparisi I think your problem might be more or less a duplicate of #6141.

**Comment by loretoparisi** [comment]
@botev thank you, that is.

**Comment by PumpkingWok** [comment]
@loretoparisi Thanks for the reply, I'll try this on the week. In the meantime I have resolved this installing `Theano 0.10 dev` with `Cuda8` and `cuDNN6`. Thank again !!
